crates/db-device/src/lib.rs

```diff
@@ -0,0 +1,436 @@
+// Copyright 2020 Alibaba Cloud. All Rights Reserved.
+// Copyright © 2019 Intel Corporation. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+#![deny(missing_docs)]
+
+//! dragonball-vmm device model.
+
+pub mod device_manager;
+pub mod resources;
+
+use std::cmp::{Ord, PartialOrd};
+use std::sync::Mutex;
+
+use self::resources::DeviceResources;
```

Comment:
Is the `self` here needed?

Reply:
it's option and good for 2018 to avoid name conflicting.

Reply:
Got that